src/main/kotlin/org/rust/ide/colorscheme/RustColorSettingsPage.kt

```diff
@@ -5,46 +5,21 @@ import com.intellij.openapi.options.colors.ColorSettingsPage
 import org.rust.ide.highlight.syntax.RustHighlighter
 import org.rust.ide.icons.RustIcons
 import org.rust.ide.utils.loadCodeSampleResource
-import com.intellij.openapi.options.colors.AttributesDescriptor as d
 
 class RustColorSettingsPage : ColorSettingsPage {
-    private val ATTRS = arrayOf(
-        d("Identifier", RustColors.IDENTIFIER),
-        d("Function declaration", RustColors.FUNCTION_DECLARATION),
-        d("Instance method declaration", RustColors.INSTANCE_METHOD),
-        d("Static method declaration", RustColors.STATIC_METHOD),
-        d("Lifetime", RustColors.LIFETIME),
-        d("Char", RustColors.CHAR),
-        d("String", RustColors.STRING),
-        d("Number", RustColors.NUMBER),
-        d("Keyword", RustColors.KEYWORD),
-        d("Block comment", RustColors.BLOCK_COMMENT),
-        d("Line comment", RustColors.EOL_COMMENT),
-        d("Doc comment", RustColors.DOC_COMMENT),
-        d("Parenthesis", RustColors.PARENTHESIS),
-        d("Brackets", RustColors.BRACKETS),
-        d("Braces", RustColors.BRACES),
-        d("Operator sign", RustColors.OPERATORS),
-        d("Semicolon", RustColors.SEMICOLON),
-        d("Dot", RustColors.DOT),
-        d("Comma", RustColors.COMMA),
-        d("Attribute", RustColors.ATTRIBUTE),
-        d("Macro", RustColors.MACRO),
-        d("Type Parameter", RustColors.TYPE_PARAMETER),
-        d("Mutable binding", RustColors.MUT_BINDING),
-        d("Valid escape sequence", RustColors.VALID_STRING_ESCAPE),
-        d("Invalid escape sequence", RustColors.INVALID_STRING_ESCAPE)
-    )
+    private val ATTRS = RustColor.values().map { it.attributesDescriptor }.toTypedArray()
+
     // This tags should be kept in sync with RustHighlightingAnnotator highlighting logic
     private val ANNOTATOR_TAGS = mapOf(
-        "attribute" to RustColors.ATTRIBUTE,
-        "macro" to RustColors.MACRO,
-        "type-parameter" to RustColors.TYPE_PARAMETER,
-        "mut-binding" to RustColors.MUT_BINDING,
-        "function-decl" to RustColors.FUNCTION_DECLARATION,
-        "instance-method-decl" to RustColors.INSTANCE_METHOD,
-        "static-method-decl" to RustColors.STATIC_METHOD
-    )
+        "attribute" to RustColor.ATTRIBUTE,
+        "macro" to RustColor.MACRO,
+        "type-parameter" to RustColor.TYPE_PARAMETER,
+        "mut-binding" to RustColor.MUT_BINDING,
+        "function-decl" to RustColor.FUNCTION_DECLARATION,
+        "instance-method-decl" to RustColor.INSTANCE_METHOD,
+        "static-method-decl" to RustColor.STATIC_METHOD
+    ).mapValues { it.component2().textAttributesKey }
```

Comment:
This line seems highly non-obvious to me (but maybe I just don't know Kotlin enough, so I had to look up all the function signatures involved).
Maybe it is better to use `it.value.textAttributesKey`?